### ISSUE TITLE
Test LMDB job automatically deleted

### ISSUE THREAD

**Issue by LucMioulet** [NVIDIA/DIGITS#1118]
Hello,
I have built my own LMDB database for classification and I have managed to train successfully models on it.
Now when I want to visualize the results using the "Test LMDB" option with a valid LMDB  it doesn't work.
The job is generated, the LMDB is parsed and the output hdf5 is created, but as soon as this is done the job is deleted.

2016-09-26 14:45:45 [20160926-144529-f2e3] [WARNING] Infer Model unrecognized output: Processed 31773/31773 images
2016-09-26 14:45:45 [20160926-144529-f2e3] [WARNING] Infer Model unrecognized output: /FOLDER/TO/20160926-144529-f2e3/inference.hdf5
2016-09-26 14:45:46 [20160926-144529-f2e3] [INFO ] Infer Model task completed.
2016-09-26 14:45:46 [20160926-144529-f2e3] [INFO ] Job complete.
2016-09-26 14:45:46 [20160926-144529-f2e3] [INFO ] Job deleted.

The GUI doesn't generate any visualization page after that.
Any idea what could go on ?

**Comment by gheinrich** [comment]
It's difficult to say, you might want to walk through `digits/tools/inference.py` and [`infer_db`](https://github.com/NVIDIA/DIGITS/blob/digits-4.0/digits/model/images/generic/views.py#L369) to see what is going on.

**Comment by lukeyeager** [comment]
Tested with DIGITS 5.0 and it works as expected. Closing.

If you can provide some more information and/or steps to reproduce, please do so and reopen this issue.

**Comment by ekcolam** [comment]
I have some more information this problem, as this was something similar we saw yesterday.
SW Installation:
- OS: RHEL 6 (kernel 2.6.32-642.1.1)
- DIGITS 4.0
- Python 2.7.10
- HDF5 1.8.17
- pyparsing == 1.5.7
- pydot == 1.0.28
- graphviz (python) 2.26.0

When attempting to run classification on one item, there appears to be an underlying conflict between the dot_parser for Python and the pyparsing/pydot package combo.  When I upgrade the pyparsing package to 2.1.10, I don't see this issue, but I do get a warning when running DIGITS that it cannot find the dot_parser binary.  Was this problem fixed in DIGITS 5?

**Comment by lukeyeager** [comment]
So, your problem has nothing to do with this bug? Is anything getting deleted?

**Comment by ekcolam** [comment]
The inference file it is saving is getting deleted according to the DIGITS log, but the Classify One option does not work.  This is what we were seeing in our DIGITS log:

> 2017-02-07 13:48:04 [20170207-134802-8115] [INFO ] Infer Model task started.
> 2017-02-07 13:48:04 [20170207-134802-8115] [INFO ] Task subprocess args: ['/usr/bin/python2', '/path/to/DIGITS/DIGITS-4.0.0/tools/inference.py', '/path/to/DIGITS/DIGITS-4.0.0/digits/jobs/20170207-134802-8115/tmpgO8Ilc.txt', '/path/to/DIGITS/DIGITS-4.0.0/digits/jobs/20170207-134802-8115', '20170207-133942-f75f', '--jobs_dir=/path/to/DIGITS/DIGITS-4.0.0/digits/jobs', '--epoch=30.0', '--layers=none', '--gpu=0']
> 2017-02-07 13:48:10 [20170207-134802-8115] [WARNING] Infer Model unrecognized output: [?1034h2017-02-07 13:48:10 [INFO ] Saved data to /path/to/DIGITS/DIGITS-4.0.0/digits/jobs/20170207-134802-8115/inference.hdf5
> 2017-02-07 13:48:10 [20170207-134802-8115] [INFO ] Infer Model task completed.
> 2017-02-07 13:48:11 [20170207-134802-8115] [INFO ] Job complete.
> 2017-02-07 13:48:11 [20170207-134802-8115] [INFO ] Job deleted.

The Classify One screen would show an error occurred and say to check the log file.  The error I saw in the log with this command looked similar to the issue originator, hence I posted my comment here.

**Comment by lukeyeager** [comment]
Hmm. Sounds like there may be a lack of proper error reporting? You fixed your underlying error by updating pyparsing/pydot, but maybe @LucMioulet had a different problem that was also not reported properly.

**Comment by ekcolam** [comment]
Not to keep hijacking this thread, I noticed pydot was updated to 1.0.29 for DIGITS 5.  Does that work with pyparsing 2.1.10?  Or is pyparsing 1.5.7 preferred?

**Comment by lukeyeager** [comment]
@ekcolam to digress further, I don't think pydot was actually updated. I just changed the pip version to match the [incorrect] ubuntu package version: https://bugs.launchpad.net/ubuntu/+source/pydot/+bug/1609585

I've never heard of anyone having pydot/pyparsing versioning problems. I guess you're getting weird versions on RedHat? Here are the versions I've got on Ubuntu 16.04 which are working well:
```sh
$ pip list | egrep -i 'dot|pars'
pydot (1.0.29)
pyparsing (2.0.3)
```
